providers/darwin/host_darwin_amd64.go

```diff
@@ -46,6 +46,87 @@ func (h *host) Info() types.HostInfo {
 	return h.info
 }
 
+func (h *host) CPUTime() (*types.CPUTimes, error) {
+	cpu, err := getHostCPULoadInfo()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get host CPU usage")
+	}
+
+	ticksPerSecond := time.Duration(getClockTicks())
+
+	return &types.CPUTimes{
+		Timestamp: time.Now(),
+		User:      time.Duration(cpu.User) * time.Second / ticksPerSecond,
+		System:    time.Duration(cpu.System) * time.Second / ticksPerSecond,
+		Idle:      time.Duration(cpu.Idle) * time.Second / ticksPerSecond,
+		Nice:      time.Duration(cpu.Nice) * time.Second / ticksPerSecond,
+	}, nil
+}
+
+func (h *host) Memory() (*types.HostMemoryInfo, error) {
+	mem := &types.HostMemoryInfo{Timestamp: time.Now()}
+
+	// Total physical memory.
+	if err := sysctlByName("hw.memsize", &mem.Total); err != nil {
+		return nil, errors.Wrap(err, "failed to get total physical memory")
+	}
+
+	// Page size for computing byte totals.
+	pageSizeBytes, err := getPageSize()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get page size")
+	}
+
+	// Virtual Memory Statistics
+	vmStat, err := getHostVMInfo64()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get virtual memory statistics")
+	}
+
+	// Swap
+	swap, err := getSwapUsage()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get swap usage")
+	}
+
+	inactiveBytes := uint64(vmStat.Inactive_count) * pageSizeBytes
+	purgeableBytes := uint64(vmStat.Purgeable_count) * pageSizeBytes
+	mem.Metrics = map[string]uint64{
+		"active_bytes":         uint64(vmStat.Active_count) * pageSizeBytes,
```

Comment:
My first thought was using `.bytes` for this but probably not the right place to do this here.

Reply:
Yep, I was thinking that also. I was giving thought to creating a metric.Counter and metric.Gauge type that held the units. That would probably make it easier to consume and automatically add `.bytes` on the Beats side.

Reply:
Lets leave it as is for now as we can adjust it on the beats side manually for now.